pkg/gomod/selector.go

```diff
@@ -0,0 +1,104 @@
+package gomod
+
+import (
+	"log"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+	"knative.dev/test-infra/pkg/gowork"
+)
+
+var (
+	// ErrNoDomain is returned when no domain is provided.
+	ErrNoDomain = errors.New("no domain provided")
+)
+
+// Matcher matches a module name.
+type Matcher interface {
+	// Match returns true if the module name matches.
+	Match(modname string) bool
+}
+
+// MatcherFunc is a function that implements the Matcher interface.
+func MatcherFunc(fn matcherFunc) Matcher {
+	return fn
+}
```

Comment:
Do we need an interface / adapter here, or would a simple type alias like `matcherFunc` do? i.e. what if we changed Selector to:

```golang

// Filter is a function which accepts or rejects a module name
type Filter func(modname tring) bool


// Selector filters modules based on a set of include and exclude functions.
type Selector struct {
  Includes []Filter
  Excludes []Filter
}
```

Reply:
I think the interface should stay, as there are 2 implementations: `matcherFunc` and `CurrentModulesMatcher` struct. The `CurrentModulesMatcher` struct holds `sync.Once` to collect project information just once.